oauth2/oauth2-client/src/test/java/org/springframework/security/oauth2/client/OAuth2AuthorizedClientSessionDestroyedListenerTests.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2002-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.security.oauth2.client;
+
+import static org.mockito.Mockito.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.when;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.context.ApplicationListener;
+import org.springframework.security.authentication.TestingAuthenticationToken;
+import org.springframework.security.core.context.SecurityContext;
+import org.springframework.security.core.context.SecurityContextImpl;
+import org.springframework.security.core.session.SessionDestroyedEvent;
+import org.springframework.security.oauth2.client.authentication.OAuth2AuthenticationToken;
+import org.springframework.security.oauth2.core.user.OAuth2User;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Tests for {@link OAuth2AuthorizedClientSessionDestroyedListener}.
+ *
+ * @author Kazuki Shimizu
+ * @since 5.0
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class OAuth2AuthorizedClientSessionDestroyedListenerTests {
+
+	@Mock
+	private OAuth2AuthorizedClientService authorizedClientService;
+
+	@Test(expected = IllegalArgumentException.class)
+	public void constructorNullOAuth2AuthorizedClientService() {
+		new OAuth2AuthorizedClientSessionDestroyedListener(null);
+	}
+
+	@Test
+	public void removesAuthorizedClient() {
```

Comment:
The test method naming convention that we are using going forward is as follows:

`methodNameWhen<condition>Then<Expectation>`

e.g. `constructorWhenPrincipalIsEmptyStringThenThrowIllegalArgumentException`

Can you please rename accordingly. Take a look at the existing tests in `oauth2-client` module for more examples.

Reply:
fix via 3c7cd39e7ed03054d039dff640c6117db7e4cec9